Gems/Atom/RPI/Code/Source/RPI.Public/Pass/CopyPass.cpp

```diff
@@ -262,119 +279,139 @@ namespace AZ
 
         void CopyPass::SetupFrameGraphDependenciesDeviceToHost(RHI::FrameGraphInterface frameGraph)
         {
-            // We need the size of the output image when copying from image to image, so we need all attachments (even the output ones)
-            // We also need it so the framegraph knows the two scopes depend on each other
-            DeclareAttachmentsToFrameGraph(frameGraph);
+            if (m_inputOutputCopy)
+            {
+                // We need to manually do what DeclareAttachmentsToFrameGraph in order to correctly set the ScopeAttachmentAccess to Read
+                // since we only read within this scope.
```

Comment:
Could we not simply pass RHI::ScopeAttachmentAccess::Read as variable into DeclareAttachmentsToFrameGraph instead of duplicating the code here?

Reply:
Implemented with a mask for the access flags.